samples/ObjectTracker/cs/ObjectTrackerSample_UWP/MainPage.xaml.cs

```diff
@@ -0,0 +1,874 @@
+        private List<List<TrackerResult>> m_trackerHistory = null;
+
+        // Frame source
+        private IFrameSource m_frameSource = null;
+        private SoftwareBitmapSource m_bitmapSource = new SoftwareBitmapSource();
+        private bool m_frameSourceIsStreaming = false;
+
+        // UI-related variables
+        private ObjectTrackRenderer m_objectTrackRenderer = null;
+        private uint m_cameraFrameWidth, m_cameraFrameHeight;
+        private bool m_isCameraFrameDimensionInitialized = false;
+
+        // Stopwatches
+        private Stopwatch m_skillEvalStopWatch = new Stopwatch();
+        private Stopwatch m_renderStopWatch = new Stopwatch();
+
+        // Synchronization
+        private SemaphoreSlim m_lock = new SemaphoreSlim(1);
+        private SemaphoreSlim m_bboxLock = new SemaphoreSlim(1);
+        private SemaphoreSlim m_renderLock = new SemaphoreSlim(1);
+
+        /// <summary>
+        /// Initializes a new instance of the MainPage class.
+        /// </summary>
+        public MainPage()
+        {
+            this.InitializeComponent();
+
+            // Set supported inking device types.
+            UIInkCanvasOverlay.InkPresenter.InputDeviceTypes =
+                Windows.UI.Core.CoreInputDeviceTypes.Mouse |
+                Windows.UI.Core.CoreInputDeviceTypes.Pen |
+                Windows.UI.Core.CoreInputDeviceTypes.Touch;
+
+            // Set initial ink stroke attributes.
+            InkDrawingAttributes drawingAttributes = new InkDrawingAttributes();
+            drawingAttributes.Size = new Size(5.0, 5.0);
+            drawingAttributes.Color = Windows.UI.Colors.Cyan;
+            drawingAttributes.IgnorePressure = false;
+            drawingAttributes.FitToCurve = true;
+            UIInkCanvasOverlay.InkPresenter.UpdateDefaultDrawingAttributes(drawingAttributes);
+            UIInkCanvasOverlay.InkPresenter.IsInputEnabled = true;
+        }
+
+        /// <summary>
+        /// Responds when we navigate to this page.
+        /// </summary>
+        /// <param name="e">Event data</param>
+        protected override async void OnNavigatedTo(NavigationEventArgs e)
+        {
+            // Initialize renderer
+            m_objectTrackRenderer = new ObjectTrackRenderer(UICanvasOverlay);
+            UICameraPreview.Source = m_bitmapSource;
+
+            // Initialize skill
+            m_descriptor = new ObjectTrackerDescriptor();
+            m_availableExecutionDevices = await m_descriptor.GetSupportedExecutionDevicesAsync();
+            if (m_availableExecutionDevices.Count == 0)
+            {
+                NotifyUser("No execution devices available, this skill cannot run on this device", NotifyType.ErrorMessage);
+                return; // Abort
+            }
+            m_skill = await m_descriptor.CreateSkillAsync() as ObjectTrackerSkill;
+            m_bindings = new List<ObjectTrackerBinding>();
+            m_trackerHistory = new List<List<TrackerResult>>();

```

Comment:
Consider refactoring into separate `InitializeSkillAsync` method and update README accordingly